### DISCUSSION TITLE
Upload file doesn't work in production / aws-s3 & Heroku

### DISCUSSION THREAD

**Discussion by sdelon** [strapi/strapi#7205] [Legacy Topics]
Hi All,

I have a problem when uploading files in production with strapi. I have deployed strapi on Heroku and I am using the upload-provider-asws-s3 for files upload. I created a settings.js file in extensions/upload/config with the following code :

```
if (process.env.NODE_ENV === 'production') {
  module.exports = {
    provider: "aws-s3",
    providerOptions: {
      accessKeyId: process.env.ACCESS_KEY_ID,
      secretAccessKey: process.env.SECRET_ACCESS_KEY,
      region: "xxxxxxxxx",
      params: {
        Bucket: "xxxxxxxxx"
      }
    }
  };
} else {
  module.exports = {}
}
```

but when I connect to the strapi production admin and try to upload files, my s3 bucket won't load the files. I connected the same aws-s3 config to the development mode (just to test) and it worked perfectly. It is like heroku won't connect to aws-s3 but I don't get what I miss in my configuration to be able to upload files in the production mode.

Thank you all for your help!

**Comment by lauriejim** [comment]
Hello! What version of Strapi are you using, if the last one (3.1.1 of at least the stable 3.0.x), you can check this docs - https://strapi.io/documentation/v3.x/plugins/upload.html#using-a-provider
This is easier to play with env on that format.

**Reply by sdelon** [reply]
Hello Jim, I am using the latest version, so I modified the path of the file as explained in the docs. I have no longer the extensions/upload/config... file but instead I created inside ./config/env/production/plugins.js and paste my configuration in there

```
module.exports = ({ env }) => ({
    upload: {
        provider: "aws-s3",
        providerOptions: {
            accessKeyId: process.env.ACCESS_KEY_ID,
            secretAccessKey: process.env.SECRET_ACCESS_KEY,
            region: "xxxxxxx",
            params: {
            Bucket: "xxxxxxx"
            },
        },
    },
});
```

but it still won't work in production mode. It works perfectly in development mode though when creating the ./config/plugins.js (I am using the same configuration in both environment). Thanks for your help !

**Reply by lauriejim** [reply]
Hello! the issue is probably in the setup of your credentials?

**Reply by sdelon** [reply]
Jim, I don't think so since it works fine in development mode. The files get uploaded on my s3-bucket but not in production. I can modify content but not files. When I try to add an img in the media library, I now have this error : "An internal server error occurred"
